### ISSUE TITLE
Timeout doesn't work when no sniffing results with L2pcapListenSocket

### ISSUE THREAD

**Issue by vorotnikovdk** [secdev/scapy#74]
OS Ubuntu 14.04.3 LTS
Scapy 2.3.2-dev
python-libpcap 0.6.4-1

I am using pylibpcap for sniffing and I notice, that timeout parameter for sniff function doesnt work if there are no recieved packets.

Example:

```
In [4]: conf.use_pcap
Out[4]: True

In [5]: conf.L2listen
Out[5]: scapy.arch.pcapdnet.L2pcapListenSocket

In [6]: sniff(iface="eth1", filter="vlan 554", timeout=3)
```
and in freezes forever if there are no packets with vlan 554.

I checked the code and found, that after this line 
https://github.com/secdev/scapy/blob/master/scapy/sendrecv.py#L584
` sel = select([s],[],[],remain)`
scapy has 
`sel[0] = [<scapy.arch.pcapdnet.L2pcapListenSocket object at 0x7f228072a310>]`
 in selected list. So function doesnt finish.

If I use `scapy.arch.linux.L2ListenSocket` everything is ok, selected list is empty.

**Comment by vorotnikovdk** [comment]
As far as I understand, _sniff()_ uses select to know if there is some packets to read.
But select call on _pcapObject_ will return readable fd even if there is no captured packets. Fd is readable if pcapObject's _next()_ call will not be blocked. It doesn't guarantee that there are some packets and _next()_'s returning valie is not None.

So this loop in `L2pcapListenSocket.recv()`:
```
while pkt is None:
    pkt = self.ins.next()
    if pkt is not None:
        ts,pkt = pkt
```
will work until receiving at least one packet. In my case (no packets because of applied BPF filter) to runs forever and sniff()'s _timeout_ parameter doesn't work.

AFIAK this problem is actual not only for pylibpcap, but also for pcapy and pypcap
Scapy needs another way to check if L2pcapListenSocket has packets to recv

**Comment by vorotnikovdk** [comment]
I created pull request. Please, review it
Notice: it doesnt deal with winpcapy

**Comment by p-l-** [comment]
Fixed by @vorotnikovdk in #75.
